src/applications/disability-benefits/all-claims/pages/paymentInformation.js

```diff
@@ -1,20 +1,49 @@
-import { fetchPaymentInformation } from '../utils';
-import AsyncDisplayWidget from '../components/AsyncDisplayWidget';
-import PaymentView from '../components/PaymentView';
-import PaymentFailureView from '../components/PaymentFailureView';
+import fullSchema from '../config/schema';
+
+const {
+  accountType,
+  accountNumber,
+  routingNumber,
+  bankName,
+} = fullSchema.properties;
 
 export const uiSchema = {
   'ui:title': 'Payment information',
-  'ui:field': 'StringField',
-  'ui:widget': AsyncDisplayWidget,
-  'ui:options': {
-    callback: fetchPaymentInformation,
-    viewComponent: PaymentView,
-    failureComponent: PaymentFailureView,
+  'ui:description':
+    'This is the bank account information we have on file for you. We’ll pay your disability benefit to this account.',
+  accountType: {
+    'ui:title': 'Account type',
+    'ui:options': {
+      widgetClassNames: 'va-select-medium-large',
+    },
+  },
+  accountNumber: {
+    'ui:title': 'Account number',
+    'ui:options': {
+      widgetClassNames: 'va-input-medium-large',
+    },
+  },
+  routingNumber: {
+    'ui:title': 'Routing number',
+    'ui:errorMessages': {
+      pattern: 'Routing number must be 9 digits',
+    },
+    'ui:options': {
+      widgetClassNames: 'va-input-medium-large',
+    },
+  },
+  bankName: {
+    'ui:title': 'Bank name',
   },
 };
 
 export const schema = {
   type: 'object',
-  properties: {},
+  required: ['accountType', 'accountNumber', 'routingNumber', 'bankName'],
```

Comment:
So this is where it gets tricky. These fields are only required _if_ they enter any information. They should be able to submit without any of this.

@goldenmeanie What's the best way to handle that? We've historically done a yes / no question that opens this section up.

If we went down that path again, we could default that question to `yes` (or whatever) to expand the bank account information fields if we've got pre-fill information for them.

Reply:
Ahh, right, I actually forgot this was meant to be optional. No expanding necessary. We can just do a ui:required that checks whether the other fields have been touched. We do this for homelessness - phone is required if there's an email, and vice-versa.